File: pytools/data/data.code.golang.server/msg_scene_vector_3.go
package proto

import (
	"packet"
)

type MsgSceneVector3 struct {
	x                        int16
	y                        int16
	z                        int16
}

func (this *MsgSceneVector3) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteInt16(this.x)
	pack.WriteInt16(this.y)
	pack.WriteInt16(this.z)

	return pack.ReadBytes()
}

func MsgSceneVector3Decode(pack *packet.Packet) *MsgSceneVector3 {
	msgSceneVector3 := &MsgSceneVector3{}

	msgSceneVector3.x = pack.ReadInt16()
	msgSceneVector3.y = pack.ReadInt16()
	msgSceneVector3.z = pack.ReadInt16()
	return msgSceneVector3
}

func (this *MsgSceneVector3) SetX(x int16) {
	this.x = x
}

func (this *MsgSceneVector3) GetX() int16 {
	return this.x
}

func (this *MsgSceneVector3) SetY(y int16) {
	this.y = y
}

func (this *MsgSceneVector3) GetY() int16 {
	return this.y
}

func (this *MsgSceneVector3) SetZ(z int16) {
	this.z = z
}

func (this *MsgSceneVector3) GetZ() int16 {
	return this.z
}

File: pytools/examples/lua53/client/protocols/MsgSceneVector3.lua
MsgSceneVector3 = {}
setmetatable(MsgSceneVector3, {__index = _G})
local _ENV = MsgSceneVector3


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = MsgSceneVector3})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.x = 0
	self.y = 0
	self.z = 0
end


function encode(self)
	pack = packet.new()
	pack:writeI16(self.x)
	pack:writeI16(self.y)
	pack:writeI16(self.z)
	return pack:readBytes()
end

function decode(self, pack)
	self.x = pack:readI16()
	self.y = pack:readI16()
	self.z = pack:readI16()
	return self
end

function getBytes(self)
	return self:encode()
end


function setX(self, x)
	self.x = x
end
function getX(self)
	return self.x
end

function setY(self, y)
	self.y = y
end
function getY(self)
	return self.y
end

function setZ(self, z)
	self.z = z
end
function getZ(self)
	return self.z
end

